atlasdb-config/src/main/java/com/palantir/atlasdb/config/AtlasDbConfig.java

```diff
@@ -313,17 +321,19 @@ private void checkNamespaceConfig() {
                 // (C* keyspace / Postgres dbName / Oracle sid). But changing the name of the TimeLock client
                 // will return the timestamp bound store to 0, so we also need to fast forward the new client bound
                 // to a value above of the original bound.
-                Preconditions.checkState(timeLockConfig.client().equals(keyValueServiceNamespace),
+                Preconditions.checkState(timeLockConfig.client().equals(Optional.of(keyValueServiceNamespace)),
                         "AtlasDB refused to start, in order to avoid potential data corruption."
                                 + " Please contact AtlasDB support to remediate this. Specific steps are required;"
                                 + " DO NOT ATTEMPT TO FIX THIS YOURSELF.");
             }
+            return keyValueServiceNamespace;
         } else if (timelock().isPresent()) {
             // Special case - empty timelock and empty namespace/keyspace does not make sense
-            boolean timelockClientNonEmpty = !timelock().get().client().orElse("").isEmpty();
-            Preconditions.checkState(timelockClientNonEmpty,
+            Preconditions.checkState(timelock().get().client().isPresent(),
```

Comment:
Think you've lost the case where the `client()` is present but is the empty String.  Was this intentional?

Reply:
An empty string here is treated as `Optional.empty()` by Jackson. Also, this is not the right place for this check, if we wanted to have constraint checks on the client string (like it does not contain an '@' or something) that should exist in `TimelockClientConfig`.

Reply:
Added a test for the empty string case.

Reply:
True, added it at the right place and a test.